clients/vault/src/oracle/collector/proof_builder.rs

```diff
@@ -42,17 +43,36 @@ pub enum ProofStatus {
 	NoTxSetFound(Slot),
 }
 
+const MAX_SLOT_TO_REMEMBER : u64 = 12;
 // handles the creation of proofs.
 // this means it will access the maps and potentially the files.
 impl ScpMessageCollector {
 	fn get_slot(&self, tx_hash: &TxHash) -> Option<Slot> {
 		self.tx_hash_map().get(tx_hash).map(|slot| *slot)
 	}
 
+	
 	/// Returns either a list of ScpEnvelopes or a ProofStatus saying it failed to retrieve a list.
 	fn get_envelopes(&self, slot: Slot) -> Result<UnlimitedVarArray<ScpEnvelope>, ProofStatus> {
-		let (envelopes, is_from_file) =
-			self._get_envelopes(slot).ok_or(ProofStatus::NoEnvelopesFound(slot))?;
+		let envelopes =
+			self._get_envelopes(slot);
+		
+		if let None = envelopes{
```

Comment:
Let's not only do this when we don't have anything but also when we do have some but not enough.

Reply:
I would suggest something like 
```
let fetch_more = if envelopes.is_none() { true } else {
let envelopes = envelopes.unwrap();
envelopes.len()  < get_min_externalized_messages(self.is_public()) 
}

if fetch_more {
... // your logic
```

Reply:
Done.